### ISSUE TITLE
Wrong method signature for ConfigService.getOrThrow with the infer property set to true

### ISSUE THREAD

**Issue by MinaFayez9** [nestjs/config#1168]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

```
interface EnvironmentVariables {
  PORT: number;
  TIMEOUT: string;
}


constructor(private configService: ConfigService<EnvironmentVariables>) {
  // port is of type EnvironmentVariables  not number
  const port = this.configService.getOrThrow('PORT', { infer: true });
}
```
After checking the `node_modules\@nestjs\config\dist\config.service.d.ts`
- Signature of `get` is `get<T = K, P extends Path<T> = any, R = PathValue<T, P>>(propertyPath: P, options: ConfigGetOptions): ExcludeUndefinedIf<WasValidated, R>;`
- Signature of `getOrThrow` is `getOrThrow<T = K, P extends Path<T> = any>(propertyPath: P, options: ConfigGetOptions): Exclude<T, undefined>;`

### Minimum reproduction code

https://stackblitz.com/edit/nestjs-issue-minimum-repro-starter-zcprps?file=README.md

### Steps to reproduce

Check `main.ts`

### Expected behavior

Signature of `getOrThrow` should be `getOrThrow<T = K, P extends Path<T> = any, R = PathValue<T, P>>(propertyPath: P, options: ConfigGetOptions): Exclude<R, undefined>;`

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Other package

"@nestjs/config

### NestJS version

9.2.1

### Packages versions

```
"@nestjs/config": "^2.2.0"
```

### Node.js version

18.13.0

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by MinaFayez9** [comment]
It's a type error not a bug and it is very clear, so this is very bureaucratic
Here is your "[minimum reproduction repository](https://stackblitz.com/edit/nestjs-issue-minimum-repro-starter-zcprps?file=README.md)"

**Comment by kamilmysliwiec** [comment]
Thank you @MinaFayez9. Would you like to create a PR for this issue? This looks like an easy fix (`get()` method works as expected)

**Comment by MinaFayez9** [comment]
@kamilmysliwiec I am always happy to help, but after checking out the repo, I found that it was already fixed 5 months ago https://github.com/nestjs/config/commit/73074929cc87dc393c209746c3b4801be74fdff2

**Comment by MinaFayez9** [comment]
I see now from npm that you never published a new time since 7 months, that's why the fixed is not live

**Comment by MinaFayez9** [comment]
So is there a known plan for when will you have a new publish/release ?

**Comment by kamilmysliwiec** [comment]
Published as 2.3.0
